### PR TITLE
Docs for subscription callback plugin

### PR DESCRIPTION
**Pull request by trevor-scheer**

### CODE REVIEW

File: docs/source/api/plugin/subscription-callback.mdx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+title: "API Reference: Subscription callback plugin"
+api_reference: true
+---
```
Comment (dariuszkuc) [inline docs/source/api/plugin/subscription-callback.mdx:4]:
```suggestion
minVersion: 4.9.0
---
```

File: docs/source/api/plugin/subscription-callback.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+title: "API Reference: Subscription callback plugin"
+api_reference: true
+minVersion: 4.9.0
+---
+
+> This protocol is currently in [preview](/resources/product-launch-stages#preview). Breaking changes might be introduced during the preview period.
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:7]: Without prior reference, "this protocol" is a little confusing.

I would put this sentence after the next one ("This document...") To maintain the importance of it, I would bold it.

```suggestion
```

File: docs/source/api/plugin/subscription-callback.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+>
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:8]:
```suggestion
```

File: docs/source/api/plugin/subscription-callback.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+> This document covers the usage of the Subscription callback plugin for use in Federation with Apollo Router. For more information about the protocol itself, see the [subscription callback protocol](/router/executing-operations/subscription-callback-protocol).
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:9]:
```suggestion
> This document covers the usage of the Subscription callback plugin for use in Apollo Federation with Apollo Router. For more information about the protocol itself, see the [subscription callback protocol](/router/executing-operations/subscription-callback-protocol).

**This protocol is currently in [preview](/resources/product-launch-stages#preview). Breaking changes might be introduced during the preview period.**
```

According to our "official verbiage", the first instance of "federation" should be "Apollo Federation", with subsequent mentions with lowercase "federation".

File: docs/source/api/plugin/subscription-callback.mdx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+## Using the plugin
+
+This article documents the options for the `ApolloServerPluginSubscriptionCallback` plugin, which you can import from `@apollo/server/plugin/subscriptionCallback`.
+
+This plugin enables your GraphQL Server to respond to subscription operations using the subscription callback protocol. This protocol is used by Apollo Router to execute subscription operations and receive updates at a URL specified by the router.
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:15]:
```suggestion
This plugin enables your GraphQL server to respond to [subscription operations](/apollo-server/data/subscriptions/) using the [subscription callback protocol](https://www.apollographql.com/docs/router/executing-operations/subscription-callback-protocol/#initialization).  Apollo Router uses this protocol to execute subscription operations and receive updates at a URL specified by the router.
```

File: docs/source/api/plugin/subscription-callback.mdx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+<tr>
+<td>
+
+###### `heartbeatIntervalMs`
+
+`number`
+</td>
+<td>
+
+Optionally configure the heartbeat interval (provided in milliseconds). The default is 5 seconds, which is the interval that Apollo Router expects. Lengthening this interval will cause Apollo Router to invalidate existing subscriptions frequently and is not recommended. You might want to shorten this interval if you have latency issues between your GraphQL Server and Apollo Router.
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:61]:
```suggestion
Optionally configure the heartbeat interval in milliseconds. The default is 5 seconds, which is the interval that Apollo Router expects. Lengthening this interval may cause Apollo Router to invalidate existing subscriptions frequently and is not recommended. You may want to shorten this interval if you have latency issues between your GraphQL Server and Apollo Router.
```

File: docs/source/api/plugin/subscription-callback.mdx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+<tr>
+<td>
+
+###### `logger`
+
+`Logger`
+</td>
+<td>
+
+Optionally provide a `Logger` instance to capture logs from the plugin.
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:75]: Is there a good link we can add to `Logger` in case folks are unfamiliar?

**Review by Meschreiber — APPROVED** [review]
Just a couple of nits, but otherwise LGTM!

Also makes me think it would be nice to display parameter/options defaults prominently if they exist. (E.g. below an option's type.) Taking a note of it...

File: docs/source/api/plugin/subscription-callback.mdx (hunk 1 above)
Comment (Meschreiber) [reply docs/source/api/plugin/subscription-callback.mdx:5]: Yesss, thanks for suggesting use of the new component @dariuszkuc ! ❤️

File: docs/source/api/plugin/subscription-callback.mdx (hunk 7 above)
Comment (trevor-scheer) [reply docs/source/api/plugin/subscription-callback.mdx]: Yep, good call! Added

File: docs/source/api/plugin/subscription-callback.mdx (hunk 2 above)
Comment (trevor-scheer) [reply docs/source/api/plugin/subscription-callback.mdx]: Agreed, much better on the next line. I added a **⚠️ Note:** in front of it. [33bb766](https://github.com/apollographql/apollo-server/pull/7666/commits/33bb7669841ccc7861e22b2c13110d31b90873a0)

File: docs/source/api/plugin/subscription-callback.mdx (hunk 3 above)
Comment (trevor-scheer) [reply docs/source/api/plugin/subscription-callback.mdx]: [346721c](https://github.com/apollographql/apollo-server/pull/7666/commits/346721cfb8d4c825563296c8e59d7fddf6a55404)

File: docs/source/api/plugin/subscription-callback.mdx [hunk 8]
```diff
@@ -0,0 +1,121 @@
+---
+title: "API Reference: Subscription callback plugin"
+api_reference: true
+minVersion: 4.9.0
+---
+
+> This document covers the usage of the Subscription callback plugin for use in Apollo Federation with Apollo Router. For more information about the protocol itself, see the [subscription callback protocol](/router/executing-operations/subscription-callback-protocol).
```
Comment (Meschreiber) [inline docs/source/api/plugin/subscription-callback.mdx:7]: This is getting wrapped so it looks a little funky:
image


Might be easier to just wrap the whole thing in a `<blockquote>`

**Review by Meschreiber — APPROVED** [review]
One last nit, but approved on my end!

**Pull request merged**